src/installer.ts

```diff
@@ -42,31 +39,62 @@ export async function getNode(
     console.log(`Found in cache @ ${toolPath}`);
   } else {
     console.log(`Attempting to download ${versionSpec}...`);
-    let info = await getInfoFromManifest(versionSpec, stable, token);
-    if (!info) {
-      console.log(
-        'Not found in manifest.  Falling back to download directly from Node'
-      );
-      info = await getInfoFromDist(versionSpec);
-    }
-
-    if (!info) {
-      throw new Error(
-        `Unable to find Node version '${versionSpec}' for platform ${osPlat} and architecture ${osArch}.`
-      );
-    }
-
-    console.log(`Acquiring ${info.resolvedVersion} from ${info.downloadUrl}`);
-
     let downloadPath = '';
+    let info: INodeVersionInfo | null = null;
+
+    //
+    // Try download from internal distribution (popular versions only)
+    //
     try {
-      downloadPath = await tc.downloadTool(info.downloadUrl, undefined, token);
+      info = await getInfoFromManifest(versionSpec, stable, token);
+      if (info) {
+        console.log(
+          `Acquiring ${info.resolvedVersion} from ${info.downloadUrl}`
+        );
+        downloadPath = await tc.downloadTool(
+          info.downloadUrl,
+          undefined,
+          token
+        );
+      } else {
+        console.log(
+          'Not found in manifest.  Falling back to download directly from Node'
+        );
+      }
     } catch (err) {
-      if (err instanceof tc.HTTPError && err.httpStatusCode == 404) {
-        return await acquireNodeFromFallbackLocation(info.resolvedVersion);
+      // Rate limited?
+      if (err instanceof tc.HTTPError && err.httpStatusCode === 403) {
+        console.log(
+          'Received HTTP status code 403.  This usually indicates the rate limit has been exceeded'
+        );
+      } else {
+        console.log(err.message);
       }
+      core.debug(err.stack);
```

Comment:
I recently started liking err.stack (prints the error message + stack trace). I dont mind yanking this if you prefer. Lmk

Reply:
It's fine.